precompile/contract_native_minter.go

```diff
@@ -0,0 +1,129 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package precompile
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var (
+	_ StatefulPrecompileConfig = &ContractNativeMinterConfig{}
+	// Singleton StatefulPrecompiledContract for W/R access to the contract deployer allow list.
+	ContractNativeMinterPrecompile StatefulPrecompiledContract = createNativeMinterPrecompile(ContractNativeMinterAddress)
+
+	mintSignature = CalculateFunctionSelector("mint(address,uint256)") // address, amount
+
+	ErrCannotMint = errors.New("non-enabled cannot mint")
+
+	inputLen = common.HashLength + common.HashLength
+)
+
+// ContractNativeMinterConfig wraps [AllowListConfig] and uses it to implement the StatefulPrecompileConfig
+// interface while adding in the contract deployer specific precompile address.
+type ContractNativeMinterConfig struct {
+	AllowListConfig
+}
+
+// Address returns the address of the contract deployer allow list.
+func (c *ContractNativeMinterConfig) Address() common.Address {
+	return ContractNativeMinterAddress
+}
+
+// Configure configures [state] with the desired admins based on [c].
+func (c *ContractNativeMinterConfig) Configure(state StateDB) {
+	c.AllowListConfig.Configure(state, ContractNativeMinterAddress)
+}
+
+// Contract returns the singleton stateful precompiled contract to be used for the allow list.
+func (c *ContractNativeMinterConfig) Contract() StatefulPrecompiledContract {
+	return ContractNativeMinterPrecompile
+}
+
+// GetContractNativeMinterStatus returns the role of [address] for the minter list.
+func GetContractNativeMinterStatus(stateDB StateDB, address common.Address) AllowListRole {
+	return getAllowListStatus(stateDB, ContractNativeMinterAddress, address)
+}
+
+// SetContractNativeMinterStatus sets the permissions of [address] to [role] for the
+// contract deployer allow list.
+// assumes [role] has already been verified as valid.
+func SetContractNativeMinterStatus(stateDB StateDB, address common.Address, role AllowListRole) {
+	setAllowListRole(stateDB, ContractNativeMinterAddress, address, role)
+}
+
+// PackMintInput packs [address] and [amount] into the appropriate arguments for minting operation.
+func PackMintInput(address common.Address, amount *big.Int) ([]byte, error) {
+	// function selector (4 bytes) + input(hash for address + hash for amount)
+	input := make([]byte, 0, selectorLen+inputLen)
```

Comment:
can we change the name of `inputLen` to be specific to the function it's used for?

Reply:
done